tools/mason/MasonNew.chpl

```diff
@@ -47,18 +51,40 @@ proc masonNew(args) throws {
         else if arg == '--show' {
           show = true;
         }
+        else if arg.startsWith('--moduleName') {
+          isModuleName = true;
+          if arg.startsWith('--moduleName=') {
+            var res = arg.split("=");
+            moduleName = res[2];
+            name = args[2];
+            break;
+          }
+          else {
+            moduleName = args[i+2];
+            name = args[2];
+            break;
+          }
+        }
         else {
           name = arg;
         }
       }
-      
-      if validatePackageName(name) {
-        if isDir(name) {
-          throw new owned MasonError("A directory named '" + name + "' already exists");
+      if isModuleName then {
+        if validatePackageName(moduleName) {
+          if isDir(name) {
+            throw new owned MasonError("A directory named '" + name + "' already exists");
+          }
+          InitProject(name, moduleName, isModuleName, vcs, show);
+        }
+      }
+      else {
+        if validatePackageName(name) {
+          if isDir(name) {
+            throw new owned MasonError("A directory named '" + name + "' already exists");
+          }
+          InitProject(name, moduleName='', isModuleName, vcs, show);
```

Comment:
It may be cleaner to always define `moduleName` and `name` regardless of whether `--moduleName` was passed.

Then we could reduce some branching in the code and call `InitProject` with both args:

```
InitProject(name, moduleName, vcs, show);
```

Also, I think `packageName` would be a more accurate identifier, since it impacts both the Chapel module name and the `[brick][name]` in `Mason.toml`. We can also consider renaming `name` to `dirName` to reduce ambiguity.

Reply:
This could also eliminate the need for `isModuleName` entirely. We would know if `--moduleName` was called by checking if `moduleName == name` (or `packageName == dirName` with my suggested renames).

Reply:
> Also, I think packageName would be a more accurate identifier, since it impacts both the Chapel module name and the [brick][name] in Mason.toml

@ben-albrecht The `--moduleName` flag should only change the `.chpl` file inside of `src/` right ? I'm unclear about what  it has to do with `[brick][name]`? Shouldn't brick-name be the same as that of the project's `cwd` ? 
For example , `mason new project-testSrc --moduleName=project` should perform as follows: 

```
project-testSrc/
    Mason.toml      # [brick][name] = "project-testSrc"
    src/
         project.chpl         # module project { ...... }
    example/
    test/
```

Reply:
I believe we want the brick name to match the Chapel module name

Reply:
> This could also eliminate the need for isModuleName entirely

@ben-albrecht  Even if we get rid of `isModuleName`, we will require it to avoid overwriting of name in the last condition as : 
```
...
 else if arg.startsWith('--moduleName') {
   isModuleName = true;
   if arg.startsWith('--moduleName=') {
     var res = arg.split("=");
     moduleName = res[2];
     name = args[2];
   }
   else {
    moduleName = args[countArgs];
    name = args[2];
   }
 }
 else {
 # Name got overwritten , need to use isModuleName check here
  name = arg;
}
...
```